src/utils.ts

```diff
@@ -0,0 +1,41 @@
+import { DataViewerProposal } from './interfaces/DataViewerProposal';
+
+/**
+   * Safely gets the proposal value for a given proposal attribute key.
+   *
+   * @param  {DataViewerProposal} proposal
+   * @param  {string} key The key for the proposal attribute we want.
+   * @return {(Array|undefined)}
+   */
+const getProposalValue = (proposal: DataViewerProposal, key: string) => {
+  // This is a temporary fix:
+  // The API currently allows fields to be set to blank strings,
+  // and dutifully includes those in the value array.
+  // We do not consider those valid when choosing a proposal value,
+  // so here we are making sure we have at least one non-blank value.
+  if (!proposal.values[key]?.some((value) => value !== '')) {
```

Comment:
We might want to [trim](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim) the values, too, in case some are whitespace only:

```suggestion
  if (!proposal.values[key]?.some((value) => value.trim() !== '')) {
```